api/src/main/java/org/apache/iceberg/geospatial/GeospatialBound.java

```diff
@@ -0,0 +1,313 @@
+ * from the regular WKB representation. Geometry and geography bounds are single point encoded as a
+ * concatenation of 8-byte little-endian IEEE 754 coordinate values in the order X, Y, Z (optional),
+ * M (optional).
+ *
+ * <p>The encoding varies based on which coordinates are present:
+ *
+ * <ul>
+ *   <li>x:y (2 doubles) when both z and m are unset
+ *   <li>x:y:z (3 doubles) when only m is unset
+ *   <li>x:y:NaN:m (4 doubles) when only z is unset
+ *   <li>x:y:z:m (4 doubles) when all coordinates are set
+ * </ul>
+ *
+ * <p>This class represents a lower or upper geospatial bound and handles serialization and
+ * deserialization of these bounds to/from byte arrays, conforming to the Iceberg specification.
+ */
+public class GeospatialBound {
+  /**
+   * Parses a geospatial bound from a byte buffer according to Iceberg spec.
+   *
+   * <p>Based on the buffer size, this method determines which coordinates are present: - 16 bytes
+   * (2 doubles): x and y only - 24 bytes (3 doubles): x, y, and z - 32 bytes (4 doubles): x, y, z
+   * (might be NaN), and m
+   *
+   * <p>The ordinates are encoded as 8-byte little-endian IEEE 754 values.
+   *
+   * @param buffer the ByteBuffer containing the serialized geospatial bound
+   * @return a GeospatialBound object representing the parsed bound
+   * @throws IllegalArgumentException if the buffer has an invalid size
+   */
+  public static GeospatialBound fromByteBuffer(ByteBuffer buffer) {
+    Preconditions.checkArgument(
+        buffer.order() == ByteOrder.LITTLE_ENDIAN, "Unsupported byte order: big endian");
+    int size = buffer.remaining();
+    Preconditions.checkArgument(
+        size == 2 * Double.BYTES || size == 3 * Double.BYTES || size == 4 * Double.BYTES,
+        "Invalid geo spatial bound buffer size: %s. Valid sizes are 16, 24, or 32 bytes.",
+        size);
+
+    if (size == 2 * Double.BYTES) {
+      // x:y format (2 doubles)
+      double coordX = buffer.getDouble();
+      double coordY = buffer.getDouble();
+      return createXY(coordX, coordY);
+    } else if (size == 3 * Double.BYTES) {
+      // x:y:z format (3 doubles)
+      double coordX = buffer.getDouble();
+      double coordY = buffer.getDouble();
+      double coordZ = buffer.getDouble();
+      return createXYZ(coordX, coordY, coordZ);
+    } else {
+      // x:y:z:m format (4 doubles) - z might be NaN
+      double coordX = buffer.getDouble();
+      double coordY = buffer.getDouble();
+      double coordZ = buffer.getDouble();
+      double coordM = buffer.getDouble();
+      return new GeospatialBound(coordX, coordY, coordZ, coordM);
+    }
+  }
+
+  /**
+   * Serializes this geospatial bound to a byte buffer according to Iceberg spec.
+   *
+   * <p>Following the Iceberg spec, the bound is serialized based on which coordinates are set: -
+   * x:y (2 doubles) when both z and m are unset - x:y:z (3 doubles) when only m is unset -
```

Comment:
nit: similarly use bullet list for readability

Reply:
Changed to bullet lists